### PR TITLE
fix for sdk linter retuning false warning

### PR DESCRIPTION
**Pull request by merit-maita**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6738

## Description
 - Fixed an issue where **lint** falsely warned of using `demisto.results` command, for files who have to use it, for example scripts/integrations with `indicator- format` tag, they can't use `CommandResults`, so they have to use `demisto.results`.

### CODE REVIEW

File: demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py [hunk 1]
```diff
@@ -151,9 +155,15 @@ def _demisto_results_checker(self, node):
         Adds the relevant error message using `add_message` function if one of the above exists.
         """
         try:
-            if node.func.attrname == "results" and node.func.expr.name == "demisto":
-                self.add_message("demisto-results-exists", node=node)
-
+            file_path = node.root().file.replace(".py", ".yml")
+            with open(file_path) as file_obj:
+                loaded_file_data = yaml.load(file_obj)  # type: ignore
```
Comment (dorschw) [inline demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py:160]: do not use blank `type:ignore`s, specify the error code.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 ## Unreleased
+- Fixed an issue in *lint* where it returned warnings for some unrelated files.
```
Comment (dorschw) [inline CHANGELOG.md:3]: what unrelated files?

File: demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py (hunk 1 above)
Comment (merit-maita) [reply demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py:160]: ah no, i did not intend to keep it, removing

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 ## Unreleased
 * Added a check to **is_docker_image_latest_tag** to only fail the validation on non-latest image tag when the current tag is older than 3 days.
 * Fixed an issue where **upload** would not properly show the installed version in the UI.
+* Fixed an issue in *lint* where it returned a warning regarding usage of `demisto.results` function in entities that have to use it and can't use `CommandResults` instead.
```
Comment (dorschw) [inline CHANGELOG.md:5]:
```suggestion
* Fixed an issue where **lint** falsely warned of using `demisto.results`.
```

File: demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py [hunk 4]
```diff
@@ -142,6 +146,13 @@ def _return_outputs_checker(self, node):
         except Exception:
             pass
 
+    def _check_demisto_results(self, node):
+        try:
+            if node.func.attrname == "results" and node.func.expr.name == "demisto":
+                self.add_message("demisto-results-exists", node=node)
+        except Exception:
+            pass
```
Comment (DeanArbel) [inline demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py:154]: Can you please add a comment what're the expected exceptions and why you are ignoring them

**Review by DeanArbel — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py (hunk 4 above)
Comment (merit-maita) [reply demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py:154]: i did not add this exception, it was originally there,

Comment (DeanArbel) [reply demisto_sdk/commands/lint/resources/pylint_plugins/certified_partner_level_checker.py:154]: I see, it's still a bad practice. Make sure to at least add a log in case the exception is raised.

**Review by DeanArbel — APPROVED** [review]

File: demisto_sdk/tests/constants_test.py [hunk 5]
```diff
@@ -457,6 +457,12 @@
 XSOAR_LINTER_PY3_INVALID_WARNINGS = (
     f"{GIT_ROOT}/demisto_sdk/tests/test_files/invalid_py3_XSOARLinter_warnings.py"
 )
+
+XSOAR_LINTER_PY3_NO_DEMISTO_RESULTS_WARNINGS = f"{GIT_ROOT}/demisto_sdk/tests/test_files/py3_XSOARLinter_demisto_results_warnings.py"
+# XSOAR_LINTER_PY3_DEMISTO_RESULTS_WARNINGS_YML = (
+#     f"{GIT_ROOT}/demisto_sdk/tests/test_files/py3_XSOARLinter_demisto_results_warnings.yml"
+# )
+
```
Comment (dorschw) [inline demisto_sdk/tests/constants_test.py:465]: what's this?
